### PR TITLE
:wheelchair: fix create new post and close labels

### PR DESCRIPTION
**Pull request by uiboy**
Fixes Thommys recent complaints about screen reader not responding to "create new post" and close labels on many create/edit/save screens, fixed this so that it makes it way to the next release. 

We should do another comprehensive PR perhaps with an extensive accessibility review and provide more contextual labels on all our action buttons.

### CODE REVIEW

File: lib/pages/home/modals/create_post/create_post.dart [hunk 1]
```diff
@@ -182,7 +182,7 @@ class CreatePostModalState extends State<CreatePostModal> {
 
     return OBThemedNavigationBar(
       leading: GestureDetector(
-        child: const OBIcon(OBIcons.close),
+        child: const OBIcon(OBIcons.close, semanticLabel: 'Close create new post'),
```
Comment (lifenautjoe) [inline lib/pages/home/modals/create_post/create_post.dart]: This should be localised right?

File: lib/widgets/icon.dart [hunk 2]
```diff
@@ -95,6 +96,7 @@ class OBIcon extends StatelessWidget {
                 iconData.nativeIcon,
                 size: iconSize,
                 color: iconColor,
+                semanticLabel: semanticLabel ?? 'Close',
```
Comment (lifenautjoe) [inline lib/widgets/icon.dart]: Uhh no.. this way all icons in the app will be labeled as `'Close'`.

Don't set anything by default and whatever used, pls localise.

File: lib/pages/home/modals/create_post/create_post.dart (hunk 1 above)
Comment (uiboy) [reply lib/pages/home/modals/create_post/create_post.dart]: Yes, but localising here might cause conflicts, remember we kinda decided to localise on develop or on a release branch for all PRs.

File: lib/widgets/icon.dart (hunk 2 above)
Comment (uiboy) [reply lib/widgets/icon.dart]: Oh shit, thought i was only setting it for the close button, my bad.

**Comment by uiboy** [comment]
@lifenautjoe was something missed here, this can be merged right?

**Pull request merged**